### PR TITLE
/dexsearch: improve input validation

### PR DESCRIPTION
**Pull request by SolarisFox**

### CODE REVIEW

File: chat-plugins/datasearch.js [hunk 1]
```diff
@@ -88,10 +88,11 @@ exports.commands = {
 	},
 
 	dexsearchhelp: ["/dexsearch [parameter], [parameter], [parameter], ... - Searches for Pok\u00e9mon that fulfill the selected criteria",
-		"Search categories are: type, tier, color, moves, ability, gen, recovery, priority, stat.",
+		"Search categories are: type, tier, color, moves, ability, gen, resistes, recovery, priority, stat.",
```
Comment (Zarel) [inline chat-plugins/datasearch.js]: >resistes

Comment (SolarisFox) [inline chat-plugins/datasearch.js]: I wonder who's worse at reading: myself or an average lobby user.

I vote myself.

File: chat-plugins/datasearch.js [hunk 2]
```diff
@@ -88,10 +88,11 @@ exports.commands = {
 	},
 
 	dexsearchhelp: ["/dexsearch [parameter], [parameter], [parameter], ... - Searches for Pok\u00e9mon that fulfill the selected criteria",
-		"Search categories are: type, tier, color, moves, ability, gen, recovery, priority, stat.",
+		"Search categories are: type, tier, color, moves, ability, gen, resists, recovery, priority, stat.",
 		"Valid colors are: green, red, blue, white, brown, yellow, purple, pink, gray and black.",
 		"Valid tiers are: Uber/OU/BL/UU/BL2/RU/BL3/NU/BL4/PU/NFE/LC/CAP.",
 		"Types must be followed by ' type', e.g., 'dragon type'.",
+		"A typing following 'resists' will show pokemon that resist that typing, e.g., 'resists normal'.",
```
Comment (Zarel) [inline chat-plugins/datasearch.js]: >following

(you're looking for "preceding")

Comment (Zarel) [inline chat-plugins/datasearch.js]: This phrasing is kind of awkward, let's go with `'resists' followed by a type`

**Comment by SolarisFox** [comment]
Updated.

**Pull request merged**